### PR TITLE
feat: add a speedy block for pathways

### PR DESCRIPTION
**Pull request by sekwah41**

### CODE REVIEW

File: src/main/java/com/lovetropics/extras/block/SpeedyZone.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import net.minecraft.world.level.block.Block;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.phys.shapes.VoxelShape;
+
+public class SpeedyZone extends CustomShapeBlock {
+
+    private static final VoxelShape SMALL_SHAPE = Block.box(0.0D, 0.0D, 0.0D, 16.0D, 0.5D, 16.0D);
+
+    public SpeedyZone(Properties properties) {
+        super(SMALL_SHAPE, properties.noTerrainParticles());
```
Comment (sekwah41) [inline src/main/java/com/lovetropics/extras/block/SpeedyZone.java:19]: Smol

File: src/main/java/com/lovetropics/extras/block/SpeedyZone.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    @Override
+    public void entityInside(BlockState state, Level worldIn, BlockPos pos, Entity entity) {
+        if(entity instanceof LivingEntity livingEntity) {
+            if(entity.isSteppingCarefully()) {
+                return;
+            }
+            livingEntity.addEffect(new MobEffectInstance(MobEffects.MOVEMENT_SPEED, 20, 2, false, false));
```
Comment (Gegy) [inline src/main/java/com/lovetropics/extras/block/SpeedyZone.java:28]: Would it make sense for us to do the same that we do for other SpeedyBlocks?

**Review by Gegy — APPROVED** [review]

File: src/main/java/com/lovetropics/extras/block/SpeedyZone.java (hunk 2 above)
Comment (sekwah41) [reply src/main/java/com/lovetropics/extras/block/SpeedyZone.java:28]: I believe so. The speed multiplier code sadly only worked on blocks thicker than 0.2 due to the way it detects what block type you are standing on for the step on event. And using the same multiplier code here causes rapid acceleration in an unintended way :<

Comment (sekwah41) [reply src/main/java/com/lovetropics/extras/block/SpeedyZone.java:28]: I'll add a separate pr

Comment (sekwah41) [reply src/main/java/com/lovetropics/extras/block/SpeedyZone.java:28]: Actually i wont, i want to look how to not effect the fov realistically.

**Pull request merged**